### ISSUE TITLE
`oval()` fails all tests on `wasm4-test-cart`

### ISSUE THREAD

**Issue by grantshandy** [wasmstation/wasmstation#36]
Our current `oval()` function fails *all 1529* assertions from [wasm4-test-cart](https://github.com/JerwuQu/wasm4-test-cart). wasm4-native and wasm4-web fail none:

[oval.txt](https://github.com/wasmstation/wasmstation/files/11024941/oval.txt)

I haven't looked at the code but I doubt it's the algorithm; maybe something got broken somewhere in the chain of commits.

**Comment by upachler** [comment]
do the tests pass for the original w4?

**Comment by grantshandy** [comment]
> do the tests pass for the original w4?

Yes, I tested the cart with the current native and web w4 runtime, they both pass 100%.

> I do remember that `oval()` changed in w4 a couple of months ago; maybe this test suite still reflects the old `oval()` implementation...

The most recent version of the test suite was updated for the new `oval()` (atleast that's what the commit message says). I remember you were involved in the conversation about the new `oval()` function so I thought our implementation was up to speed.

I have some time today, I'll check between the implementations for the differences.
